File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        n = len(chars)
        ans = 0
        i = 0

        while (i < n):
            letter = chars[i]
            count = 0
            
            while (i < n and chars[i] == letter):
                count += 1
                i += 1

            chars[ans] = letter
            ans += 1

            if (count > 1):
                for j in str(count):
                    chars[ans] = j
                    ans += 1

        return ans

File: 443-string-compression/string-compression.cpp
#define ll long long
#define vi vector<int>
#define vll vector<ll>
#define vs vector<string>
#define mpi map<int, int>
#define mps map<string, int>
#define pb push_back
#define all(x) (x).begin(), (x).end()

const int MOD = 1e9 + 7;

class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        int ans = 0, i = 0;

        while (i < n) {
            char ch = chars[i];
            int count = 0;

            while (i < n && chars[i] == ch) {
                count++;
                i++;
            }

            if (count == 1) {
                chars[ans++] = ch;
            } else {
                chars[ans++] = ch;
                string str = to_string(count);

                for (auto digit : str) {
                    chars[ans++] = digit;
                }
            }
        }

        return ans;
    }
};